numpy/__init__.pyi

```diff
@@ -3344,6 +3339,7 @@ class errstate(Generic[_CallType], ContextDecorator):
         traceback: None | TracebackType,
         /,
     ) -> None: ...
+    def __call__(self, func: Callable[..., Any]) -> Callable[..., Any]: ...
```

Comment:
```suggestion
    def __call__(self, func: _CallType) -> _CallType: ...
```
Use the type variable here so that the signatures of `func` and the return type match exactly.

Reply:
Thanks! This one was what i was most wondering.

Also changed the `__slots__` to be a tuple.